source/security.txt

```diff
@@ -0,0 +1,152 @@
+.. _c-security:
+
+================
+Secure Your Data
+================
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+.. facet::
+   :name: genre
+   :values: reference
+
+.. meta::
+   :keywords: ldap, authorize, ecs, aws, authenticate
+   :description: Learn how to use the C driver to secure your data.
+
+.. toctree::
+   :titlesonly:
+   :maxdepth: 1
+
+   /security/authentication
+   /security/enterprise-authentication
+
+Overview
+--------
+
+MongoDB supports multiple mechanisms that you can use to authenticate your application.
+This page contains code examples that demonstrate each of these mechanisms.
+
+.. tip::
+
+   To learn more about any of the mechanisms shown on this page, see the link
+   provided in each section.
+
+To use an authentication example from this page, copy the code example into the
+:ref:`sample application <c-auth-sample>` or your own application.
+Make sure to replace all placeholders in the code examples, such as ``<hostname>``, with
+the relevant values for your MongoDB deployment.
+
+.. _c-auth-sample:
+
+.. include:: /includes/usage-examples/sample-app-intro.rst
+
+.. literalinclude:: /includes/usage-examples/connect-sample-app.c
+   :language: c
+   :copyable: true
+   :linenos:
+   :emphasize-lines: 14-16
+
+SCRAM-SHA-256
+-------------
+
+The following code shows how to authenticate by using the ``SCRAM-SHA-256``
+authentication mechanism:
+
+.. literalinclude:: /includes/security/authentication.c
+   :language: c
+   :copyable: true
+   :start-after: // start-scram-sha-256
+   :end-before: // end-scram-sha-256
+
+To learn more about SCRAM-SHA-256 authentication, see :ref:`c-scram-sha-256` in
+the Authentication guide.
+
+SCRAM-SHA-1
+-----------
+
+The following code shows how to authenticate by using the ``SCRAM-SHA-1``
+authentication mechanism:
+
+.. literalinclude:: /includes/security/authentication.c
+   :language: c
+   :copyable: true
+   :start-after: // start-scram-sha-1
+   :end-before: // end-scram-sha-1
+
+To learn more about SCRAM-SHA-1 authentication, see :ref:`c-scram-sha-1` in
+the Authentication guide.
+
+MONGODB X.509
+-------------
+
+The following code shows how to create a connection URI to authenticate by using
+the ``X.509`` authentication mechanism:
+
+.. literalinclude:: /includes/security/authentication.c
```

Comment:
Suggest removing `:<percent-encoded password>` from the X509 example code. A username may be optionally passed, but a password is rejected.

The [auth spec](https://github.com/mongodb/specifications/blob/c84488cb9cf185e31869547a6378a701124b4665/source/auth/auth.md#mongodb-x509) notes "password MUST NOT be specified." for MONGODB-X509. Using the URI `mongodb://u:p@host/?authMechanism=MONGODB-X509` with the C driver results in this error:
```
Error parsing URI: ''MONGODB-X509' authentication mechanism does not accept a password'
```